librz/util/sdb/src/sdb.c

```diff
@@ -596,168 +600,186 @@ RZ_API int sdb_set(Sdb *s, const char *key, const char *val, ut32 cas) {
 	return sdb_set_internal(s, key, (char *)val, 0, cas);
 }
 
-static bool sdb_foreach_list_cb(void *user, const char *k, const char *v) {
-	SdbList *list = (SdbList *)user;
-	SdbKv *kv = RZ_NEW0(SdbKv);
-	/* seems like some k/v are constructed in the stack and cant be used after returning */
-	kv->base.key = strdup(k);
-	kv->base.value = strdup(v);
-	ls_append(list, kv);
-	return 1;
+static bool sdb_get_kv_list_cb(void *user, const char *k, ut32 klen, const char *v, ut32 vlen) {
+	RzPVector *list = (RzPVector *)user;
+	SdbKv *kv = sdbkv_new2(k, klen, v, vlen);
+	if (!kv) {
+		return false;
+	}
+	if (!rz_pvector_push(list, kv)) {
+		sdbkv_free(kv);
+		return false;
+	}
+	return true;
 }
 
-static int __cmp_asc(const void *a, const void *b) {
+static int __cmp_asc(const void *a, const void *b, RZ_UNUSED void *user) {
 	const SdbKv *ka = a, *kb = b;
 	return strcmp(sdbkv_key(ka), sdbkv_key(kb));
 }
 
-RZ_API SdbList *sdb_foreach_list(Sdb *s, bool sorted) {
-	SdbList *list = ls_newf((SdbListFree)sdbkv_free);
-	sdb_foreach(s, sdb_foreach_list_cb, list);
+/**
+ * \brief Get list of all key-value pairs in DB
+ * \param s DB
+ * \param sorted If set to true sort the resulting list by key in lexicographic order
+ */
+RZ_API RZ_OWN RzPVector /*<SdbKv *>*/ *sdb_get_kv_list(RZ_NONNULL Sdb *s, bool sorted) {
```

Comment:
Please rename the function to represent it's a vector, not a list

Reply:
`sdb_get_kv_vector` sounds not good too. What do you think about `sdb_get_items`?

Reply:
@pelijah yes, `sdb_get_items` sounds good.